### ISSUE TITLE
Python config scripts won't run

### ISSUE THREAD

**Issue by embeddedpenguin** [MicrochipTech/gcp-iot-core-examples#1]
I'm trying to set up my board for use on the cloud, and the scripts to config and generate my key don't want to run.

When running at_config.py:
`Traceback (most recent call last):
  File "at_config.py", line 131, in <module>
    configure_device(int(args.i2c,16))
  File "at_config.py", line 30, in configure_device
    dev.open()
  File "C:\Users\...\Documents\Atmel Studio\7.0\gcp-iot-core-examples\scripts\at_
kit_base.py", line 41, in open
    return self.device.open(vendor_id, product_id)
  File "hid.pyx", line 66, in hid.device.open
OSError: open failed`

When running at_pubkey.py

`Traceback (most recent call last):
  File "at_config.py", line 131, in <module>
    configure_device(int(args.i2c,16))
  File "at_config.py", line 30, in configure_device
    dev.open()
  File "C:\Users\...\Documents\Atmel Studio\7.0\gcp-iot-core-examples\scripts\at_
kit_base.py", line 41, in open
    return self.device.open(vendor_id, product_id)
  File "hid.pyx", line 66, in hid.device.open
OSError: open failed`

I've already run pip install -r requirements.txt 

What am I doing wrong?

**Comment by bryan-hunt** [comment]
You'll have to program your target with the firmware first then attach the "Target USB" port to the system you're running the scripts from.

If the system you're running the scripts from is linux then you'll have to set up some udev rules or run the scripts as sudo. We can cover the udev rules if that's the case.

**Comment by embeddedpenguin** [comment]
I've upgraded the firmware to 19.5.4, but I don't quite understand what you mean by "port to the system you're running the scripts from." If you mean what I think you mean, you're saying I simply need a usb from my computer to the target usb port on the xplaind. If that's the case, something is wrong because I'm doing that.

If you mean I must attach the port in cmd to the script I'm running, I'm not quite sure how I would do that.

Sorry if this is painfully obvious. The certificate process is quite confusing to me. (Also, I am on Windows so no issue there)

**Comment by bryan-hunt** [comment]
Firmware 19.5.4 is the winc1500 firmware - but now you'll need the target firmware (i.e. one of the projects from the boards directory). Once the target firmware is running a usb device will enumerate that the scripts can communicate with.

**Comment by embeddedpenguin** [comment]
Got it! It worked. Thanks a bunch

**Comment by danielepsv** [comment]
Hello,
I have the ATCRYPTOAUTH_XPRO-B.
I updated the WINC firmware (v19.6.1) with serial bridge (SAMD21 + ATWINC1500 on ext1).
Then I switch the WINC1500 to ATECC608A on ext2 (ext1 is empty -> SAMD21 + ATEC608A on ext2), I build the gcp_iot_core_example and program the SAMD21.
But when I run the script at_config.py I got the error:

**Traceback (most recent call last):
  File "at_config.py", line 131, in <module>
    configure_device(int(args.i2c,16))
  File "at_config.py", line 30, in configure_device
    dev.open()
  File "D:\IOT_PROJECT\SAMD21\gcp-iot-core-examples-master\scripts\at_kit_base.py", line 41, in open
    return self.device.open(vendor_id, product_id)
  File "hid.pyx", line 66, in hid.device.open
OSError: open failed**

What's wrong? I'm on windows 10.

Best regards,
Daniele
